lua/telescope/make_entry.lua

```diff
@@ -178,12 +178,25 @@ function make_entry.gen_from_buffer(opts)
       return nil
     end
 
+    local make_display = function(line)
+      local name = line
+      if opts.shorten_path then
+        name = utils.path_shorten(line)
+      end
+
+      local display = string.format("%" .. opts.bufnr_width ..
+                                    "s : %s", bufnr_str, name)
+      return display
+    end
+
+    local dsp = make_display(bufname)
+
     return {
       valid = true,
 
       value = bufname,
-      ordinal = bufnr_str .. " : " .. bufname,
-      display = bufnr_str .. " : " .. bufname,
+      ordinal = dsp,
+      display = dsp,
```

Comment:
display can actually just take the function itself, it allows functions to be passed to it, so as to not call display if it's expensive for every line (since sometimes they will be hidden).

So you can actually just say `display = make_display` here.

Reply:
Ah okay. Thanks. I am slowly getting the hang of it.

I'm not sure what to set `ordinal` to because you always set `ordinal` to `line` when you use `shorten_path`.